framework/core/js/src/forum/states/PostStreamState.ts

```diff
@@ -1,55 +1,71 @@
 import app from '../../forum/app';
 import { throttle } from 'throttle-debounce';
 import anchorScroll from '../../common/utils/anchorScroll';
+import Discussion from '../../common/models/Discussion';
+import Post from '../../common/models/Post';
 
-class PostStreamState {
-  constructor(discussion, includedPosts = []) {
-    /**
-     * The discussion to display the post stream for.
-     *
-     * @type {Discussion}
-     */
-    this.discussion = discussion;
+export default class PostStreamState {
+  /**
+   * @see https://github.com/Microsoft/TypeScript/issues/3841#issuecomment-337560146
+   */
+  ['constructor']: typeof PostStreamState;
```

Comment:
Hah, clever!

Reply:
All credit goes to the one guy on stack overflow :wink: